src/index_runner/event_loop.py

```diff
@@ -0,0 +1,124 @@
+'''
+The event loop for the index runner.
+'''
+
+import json
+import logging
+import os
+import requests
+import time
+import traceback
+
+from logging import Logger
+from typing import Callable, Dict, Any
+
+from confluent_kafka import Consumer, KafkaError
+
+from src.utils.config import config
+from src.utils.service_utils import wait_for_dependencies
+
+Message = Dict[str, Any]
+
+# TODO TEST unit tests
```

Comment:
I'll add some tests once people have taken a look at this and are ok with the approach.